libs/gson-support/src/main/java/org/codice/gsonsupport/GsonTypeAdapters.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.gsonsupport;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.Nullable;
+import org.apache.commons.collections4.map.LinkedMap;
+
+/**
+ * Utility class to generate Gson instances customized to work with our expected input/output
+ * formats.
+ */
+public class GsonTypeAdapters {
+  public static final Type MAP_STRING_TO_OBJECT_TYPE =
+      new TypeToken<Map<String, Object>>() {}.getType();
+
+  public static final Type LIST_STRING = new TypeToken<List<String>>() {}.getType();
+
+  private GsonTypeAdapters() {}
+
+  /** Adapter to mimic our old Boon-style numeric representations with Gson. */
+  public static class DdfNumericTypeAdapter extends TypeAdapter<Object> {
```

Comment:
✏️ Do not include `DDF` in class names.  If this is actually DDF specific then move its package to `org.codice.ddf` but this does not look DDF specific.  `LongAndDoubleTypeAdapter` would tell more about what the `TypeAdapter` actually does.